src/main/java/uk/ac/ebi/eva/pipeline/runner/JobExecutionApplicationListener.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015-2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.runner;
+
+import org.springframework.batch.core.BatchStatus;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.boot.autoconfigure.batch.JobExecutionEvent;
+import org.springframework.context.ApplicationListener;
+
+/**
+ * This class stores the last job execution state
+ */
+public class JobExecutionApplicationListener implements ApplicationListener<JobExecutionEvent> {
+
+    private JobExecution execution;
+
+    @Override
+    public void onApplicationEvent(JobExecutionEvent event) {
+        execution = event.getJobExecution();
+    }
+
+    public boolean isJobExecutionComplete() {
+        return execution != null
+                && execution.getStatus().compareTo(BatchStatus.COMPLETED) == 0
+                && execution.getExitStatus().compareTo(ExitStatus.COMPLETED) == 0;
+    }
```

Comment:
First one can be changed for something better the second one no, It's equal function is... funny at best.